test/e2e/clusterclass_changes_test.go

```diff
@@ -43,6 +43,15 @@ var _ = Describe("When testing ClusterClass changes [ClusterClass]", func() {
 					"spec.template.spec.numCPUs": int64(4),
 				},
 			}
+			if testMode == GovmomiTestMode {
+				clusterClassChangesSpecInput.ModifyMachineDeploymentInfrastructureMachineTemplateFields = map[string]interface{}{
+					"spec.template.spec.numCPUs": int64(4),
+				}
+			}
+			clusterClassChangesSpecInput.ModifyMachineDeploymentInfrastructureMachineTemplateFields = map[string]interface{}{
+				"spec.template.spec.powerOffMode": "hard",
```

Comment:
```suggestion
				"spec.template.spec.powerOffMode": "trySoft",
```

hard is default, so setting it to hard is a no-op

Reply:
Let's please add an if around this statement (like above just with SupervisorTestMode)

Otherwise ModifyMachineDeploymentInfrastructureMachineTemplateFields will always set powerOffMode

Reply:
Fixed it!

Reply:
Thx!